### DISCUSSION TITLE
get open_mfdataset to return numpy arrays

### DISCUSSION THREAD

**Discussion by raybellwaves** [pydata/xarray#5689] [Q&A]
I'm using `open_mfdataset` on many zarr stores which each have a large number of chunks.

I would like to return a Dataset where the data_var is a numpy array i.e. bring it all into memory in the `open_mfdataset` call but i'm having difficulties. I've played around the `chunks` arg and `parallel` arg but they all seem return a `dask.array`. It could be the chunking on the underlying zarr stores which require loading. Here's some code to test:

### Generate zarr stores
```
import xarray as xr
from tqdm.auto import tqdm


ds = xr.tutorial.open_dataset("air_temperature")
for i in tqdm(range(30)):
    _ds = ds.isel(time=i).expand_dims("time")
    _ds = _ds.chunk({"lat": 4, "lon": 8, "time": 1})
    _ds["air"].encoding = {}
    _ds.to_zarr(f"/tmp/_{i}.zarr", consolidated=True, mode="w")
```
### Read zarr stores

```
expected = xr.open_mfdataset("/tmp/_*.zarr", engine="zarr", chunks={}, parallel=True).load()
expected
```
![Screen Shot 2021-08-10 at 11 10 25 PM](https://user-images.githubusercontent.com/17162724/128963655-79f6f6f7-7d12-4703-8530-5cfdefec0c9d.png)

```
actual = xr.open_mfdataset("/tmp/_*.zarr", engine="zarr", chunks=None, parallel=False)
actual
```
![Screen Shot 2021-08-10 at 11 11 07 PM](https://user-images.githubusercontent.com/17162724/128963704-5e15aef2-aa3d-4598-ac9c-a23d2f66f893.png)

**Comment by Illviljan** [comment]
To open without dask you have to pass `chunks=None` to `open_dataset`. But in this line in `open_mfdataset` chunks is changed to dict:
https://github.com/pydata/xarray/blob/4bb9d9c6df77137f05e85c7cc6508fe7a93dc0e4/xarray/backends/api.py#L899

Not sure why that is.

Maybe it's possible to get around this using `preprocess` to trigger a `.compute()`?

**Reply by mathause** [reply]
or call `.load()`
